octoprint_homeassistant/__init__.py

```diff
@@ -466,11 +483,17 @@ def _generate_connection_status(self):
 
         state, _, _, _ = self._printer.get_current_connection()
         state_connected = "Disconnected" if state == "Closed" else "Connected"
-        self.mqtt_publish(
-            self._generate_topic("hassTopic", "Connected", full=True),
-            state_connected,
-            allow_queueing=True,
-        )
+        # Function can be called by on_event before on_after_startup has run.
+        # This will throw a TypeError since self.mqtt_publish is still null.
+        try:
+            self.mqtt_publish(
+                self._generate_topic("hassTopic", "Connected", full=True),
+                state_connected,
+                allow_queueing=True,
+            )
+        except TypeError:
+            self._logger.warning("mqtt_publish helper is not yet defined, not publishing connection state " + state_connected)
```

Comment:
If this is a normal sequence of events, I think we can omit the warning as it doesn't really allude to any concerns. Just check self.mqtt_publish is not none at the top of the function and return in the case that it's too early. Keep the comment though so it's clear to others why that happens.